modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -382,32 +382,43 @@ void WindowContextBase::process_mouse_scroll(GdkEventScroll* event) {
     jdouble dx = 0;
     jdouble dy = 0;
 
-    // converting direction to change in pixels
-    switch (event->direction) {
-#if GTK_CHECK_VERSION(3, 4, 0)
-        case GDK_SCROLL_SMOOTH:
-            //FIXME 3.4 ???
-            break;
+    bool is_smooth = false;
+
+#ifdef GLASS_GTK3
+        if (gdk_window_get_events(gdk_window) & GDK_SMOOTH_SCROLL_MASK) {
+            is_smooth = true;
+            if (event->direction == GDK_SCROLL_SMOOTH) {
```

Comment:
When would `GDK_SMOOTH_SCROLL_MASK` be set and direction not equal to `GDK_SCROLL_SMOOTH`? In this case, you will use a dx and dy value of 0 (which will at least now be ignored with your fix). Is this correct?